tests/CMakeLists.txt

```diff
@@ -24,7 +24,7 @@ SET (librepotest_SRCS
 ADD_EXECUTABLE(test_main ${librepotest_SRCS})
 TARGET_LINK_LIBRARIES(test_main
     librepo
-    ${CHECK_LIBRARY}
+    ${LIBCHECK_LDFLAGS}
```

Comment:
Ok, just one minor thing, we use the `_LIBRARIES` variables in other places (i.e. `LIBCHECK_LIBRARIES`), `TARGET_LINK_LIBRARIES` supports both formats. Can you change this for consistency sake? I've tested it, works on my end. Thanks!

Reply:
I was still cross-checking this across distros, but `_LIBRARIES` does not seem to work at least on Debian:
```
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/10/../../../x86_64-linux-gnu/libcheck_pic.a(check_pack.o): undefined reference to symbol '__pthread_unregister_cancel@@GLIBC_2.3.3'
/usr/bin/ld: /lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

Even if the .pc file there looks fine:
```
Requires: libsubunit
Libs: -L${libdir} -lcheck_pic  -pthread  -lrt -lm 
Libs.private:
Cflags: -I${includedir} -pthread
```

Reply:
I guess the `-pthread`  is the one to blame,as `${LIBCHECK_LIBRARIES}` plus `${LIBCHECK_LDFLAGS_OTHER}` seem to work on both Centos7 and Debian buildroots. I've amended it that way, but let me know if you prefer some other arrangement.

Reply:
Yeah, I was just composing a reply along those lines, though I can't reproduce your issue on Fedora with dymanic linking. I wonder what's the correct usage of these variables, but it seems the `_LDFLAGS` variable should be used here, as it contains all the required flags in one. Sorry about the hassle, could you put it back? :slightly_smiling_face: Thank you.

Reply:
No worries at all, you were actually too quick and caught me while still in the middle of testing my changes :smile:. Amended.